query-scope-service/src/main/java/uk/gov/gchq/palisade/service/queryscope/request/QueryScopeResponse.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.queryscope.request;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.resource.LeafResource;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+/**
+ * Represents the original data that has been sent from the client to Palisade Service for a request to access data.
+ * This data will be forwarded to a set of services with each contributing to the processing of this request.
+ * This version represents the output for query-scope-service with the filtered version of the request.
+ * Next in the sequence will be the input for the result-service which will construct a response to the client.
+ * Note there are two classes that effectively represent the same data but represent a different stage of the process.
+ * uk.gov.gchq.palisade.service.queryscope.request.QueryScopeResponse is the output from the query-scope-service.
+ * uk.gov.gchq.palisade.service.results.request.ResultsRequest is the input for the results-service.
+ */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public final class QueryScopeResponse {
```

Comment:
Which response is this? the one that outputs to the authorised data request store (redis) or to the Results service/controller?